### PR TITLE
ghq 0.6 (new formula)

### PR DESCRIPTION
**Pull request by Decors**

### CODE REVIEW

File: Library/Formula/ghq.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    ENV["GOPATH"] = buildpath
+    ENV.append_path "PATH", "#{ENV["GOPATH"]}/bin"
+    Language::Go.stage_deps resources, buildpath/"src"
+
+    system "go", "build", "-o", "ghq"
+    bin.install "ghq"
+  end
+
+  test do
+    system bin/"ghq", "--version"
```
Comment (bfontaine) [inline Library/Formula/ghq.rb]: Could you add a test that does something more substantial than e.g. `--version` or `--help`? Thanks!

**Comment by Decors** [comment]
Thanks. I've updated.

File: Library/Formula/ghq.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  desc "Remote repository management made easy"
+  homepage "https://github.com/motemen/ghq"
+  url "https://github.com/motemen/ghq.git", :tag => "v0.6", :revision => "b6f7aadbeb21ae18972577173ce175af83ce239d"
+  sha256 "d8ab0ef7386647b30210aabc79bf0391ea5c2887be0d4c6a3fb58b8c31eb9b24"
+
+  head "https://github.com/motemen/ghq.git", :branch => "master"
+
+  depends_on "go" => :build
+
+  option "without-completions", "Disable zsh completions"
```
Comment (mikemcquaid) [inline Library/Formula/ghq.rb]: Probably just kill this option; there's no harm in installing them.

File: Library/Formula/ghq.rb [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    bin.install "ghq"
+
+    if build.with? "completions"
+      zsh_completion.install "zsh/_ghq"
+    end
+  end
+
+  test do
+    s = shell_output("#{bin}/ghq root")
+    assert_match /#{Regexp.escape("#{testpath}/.ghq")}/, s
```
Comment (mikemcquaid) [inline Library/Formula/ghq.rb]: You can just use `"#{testpath}/.ghq"` as a string argument to save the escape. Also, might want to just add `shell_output` on the same line.

File: Library/Formula/ghq.rb [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    ENV["GOPATH"] = buildpath
+    ENV.append_path "PATH", "#{ENV["GOPATH"]}/bin"
+    Language::Go.stage_deps resources, buildpath/"src"
+
+    system "go", "build", "-o", "ghq"
+    bin.install "ghq"
+
+    if build.with? "completions"
+      zsh_completion.install "zsh/_ghq"
+    end
```
Comment (mikemcquaid) [inline Library/Formula/ghq.rb]: Even although you're removing this: in future `zsh_completion.install "zsh/_ghq" if build.with? "completions"` is a bit nicer

**Comment by bfontaine** [comment]
Could someone from @Homebrew/owners gives their :+1: / :-1: here since it’s a new formula?

**Comment by mikemcquaid** [comment]
Sorry, I was unclear before: let's install the ZSH completions but not offer an option to not install them (if that makes sense).

**Comment by DomT4** [comment]
Can you re-add the `zsh` completions here please? Thanks!

File: Library/Formula/ghq.rb [hunk 5]
```diff
@@ -0,0 +1,43 @@
+require "language/go"
+
+class Ghq < Formula
+  desc "Remote repository management made easy"
+  homepage "https://github.com/motemen/ghq"
+  url "https://github.com/motemen/ghq.git", :tag => "v0.6", :revision => "b6f7aadbeb21ae18972577173ce175af83ce239d"
+  sha256 "d8ab0ef7386647b30210aabc79bf0391ea5c2887be0d4c6a3fb58b8c31eb9b24"
+
+  head "https://github.com/motemen/ghq.git", :branch => "master"
```
Comment (DomT4) [inline Library/Formula/ghq.rb]: The `:branch` shouldn't be necessary here? We assume `master` is the default.

**Comment by DomT4** [comment]
Nearly there on this. One last style nit from the test-bot:

```
==> brew style ghq

1 file inspected, no offenses detected
Error: 1 problem in 1 formula

==> audit problems
ghq:
 * `option`(line 13) should be put before `depends_on`(line 11)
```

**Comment by DomT4** [comment]
Merged in https://github.com/Homebrew/homebrew/commit/20763411e16515dc2c0ce1b50565e7241ba761aa. Thank you for your contribution to Homebrew, we appreciate it! :smiley_cat:

**Pull request closed without merging**